### ISSUE TITLE
error: no such file or directory: 'PYTHON-NOT-FOUND' ubuntu

### ISSUE THREAD

**Issue by SergeyBridge** [catboost/catboost#264]
I am installing catboost for Python3 and  obtain such an error:
`error: no such file or directory: 'PYTHON-NOT-FOUND' `

ubuntu 16.04
python 3.6.4 Anaconda
CUDA8
GeForce GTX480
amd64

Catboost was compiled  with parameters

`find ./ -type f -name ya.make -exec sed -i 's/compute_30/compute_20/g' {} +`

then

`sudo ../../../ya make -r -DUSE_ARCADIA_PYTHON=no -DPYTHON_CONFIG=python3-config -DCUDA_ROOT=/usr/local/cuda-8.0`

ending with error
`error: no such file or directory: 'PYTHON-NOT-FOUND' `

1. With setting parameter -DPYTHON_CONFIG=python2-config the compilation process gained success, but the library couldn't be imported into python 3.6 script

**Comment by Noxoomo** [comment]
1) You shouldn't call compile anything with sudo (there no anaconda for root user)
2) Could you call python3-config  --configdir and provide output 
3) Try  to provide full path to python3-config
4) Try to add -DPYTHON_BIN=path_to_python_bin

**Comment by SergeyBridge** [comment]
@Noxoomo Thank you, Vasily, it has compiled and is working!!!

I compiled it WITHOUT sudo. All parameters leaved unchanged, so this is works for me:
`/git/catboost/catboost/python-package/catboost$ ../../../ya make -r -DUSE_ARCADIA_PYTHON=no -DPYTHON_CONFIG=python3-config -DCUDA_ROOT=/usr/local/cuda`

**Comment by SergeyBridge** [comment]
@Noxoomo I tried to run model training with my new compiled catboost.so_ library both in GPU and CPU modes.
It works well in both modes, but CPU mode is twice+ more quicker then GPU.

The question is:
Does it possible to take advantages in speeding up training catboost models with my NVIDIA Geforce GTX480 ?
`        #model = CatBoostRegressor(loss_function='RMSE',iterations=10000, depth=8, learning_rate=0.03,
        #                    od_type='Iter', od_wait=40, has_time=True, # od_pval=10e-7,
        #                     task_type='CPU', thread_count=4, rsm=0.2,
        #                     l2_leaf_reg=3,
        #                     random_seed=1000,
        #                     train_dir='./catboost_train_dir')

        model = CatBoostRegressor(loss_function='RMSE', eval_metric='RMSE',
                             iterations=2000, depth=8, learning_rate=0.03,
                             has_time=True, od_type='Iter', od_wait=40, # od_pval=10e-7,
                             task_type='GPU', thread_count=4, #rsm=0.2,
                             l2_leaf_reg=3,
                             random_seed=1000,
                             train_dir='./catboost_train_dir')



        train_pool = Pool(data= reg_features.loc[train], label = ylabel.loc[train, reg], 
                          has_header=True, cat_features=cat_features,  thread_count=4)

        eval_pool = Pool(data= reg_features.loc[may_test_per], label = ylabel.loc[may_test_per, reg], 
                         has_header=True, cat_features=cat_features,  thread_count=4)    

        june_pool = Pool(data= reg_features.loc[june_test_per], has_header=True, 
                         cat_features=cat_features,  thread_count=4)    

        model.fit(X=train_pool, use_best_model=True, eval_set=eval_pool, 
                  logging_level='Silent')
`
1. I tried consequently running the commented CPU model and then GPU model. 
2. GPU model has not have parameter rsm=.. , does it mean it is installed to 1 in GPU mode?

**Comment by Noxoomo** [comment]
1) GPU doesn't support RSM and it's always equal to 1 (so speed comparison is fair with rsm=1.0 only)
2) Could you please write full system configuration: CPU frequency, core count and model, RAM size
3) Pool size (feature count, categorical features count, lines could) would be also useful

GTX480 is very old hardware, so it's hard to say, will it give any profit at all

**Comment by SergeyBridge** [comment]
i5, 2.6GHz, 4 core
memory 133MHz, 16Gb
    
 *-cpu
          description: CPU
          product: Intel(R) Core(TM) i5 CPU         750  @ 2.67GHz
          vendor: Intel Corp.
          physical id: 4
          size: 1269MHz
          capacity: 4GHz
          width: 64 bits
          clock: 133MHz
     *-memory
          size: 16GiB

1 model:
train pool shape(46722, 231)
cat_features len = 8

600 models got with GPU in 26 hours, CPU in 12 hours

Sure,  GTX480 is an old hardware :)

**Comment by Noxoomo** [comment]
I have only to Tesla M2090 for Fermi cards family and on it i have profit compared to CPU, but i checked pools without numerical features or with --max-ctr-complexity 1.

Could you run CPU and GPU with --detailed-profile option and send logs printed after learning, may be there is some simple block, which doesn't not work well on Fermi

**Comment by SergeyBridge** [comment]
@Noxoomo  What should I do? Should I compile catboost with this option ? In a python notebook I use paramemeters, but thre is no such a parameter --detailed-profile. Could I use this value instead of Silent or Verbose in CatboostRegressor() call? Sorry for misunderstanding

**Comment by Noxoomo** [comment]
Sorry, i forgot this option is cmd-line only currently

It should be sufficient to add  detailed_profile=True line to list of options in __init__() method of CatboostRegressor (see core.py) and rebuild package

**Comment by SergeyBridge** [comment]
@Noxoomo Vasily, I' ve down such changes in core.py and rebuild the package:

`class CatBoostRegressor(CatBoost):
    """
    Implementation of the scikit-learn API for CatBoost regression.

    Parameters
    ----------
    Like in CatBoostClassifier, except loss_function, class_weights and
    classes_count

    loss_function : string, [default='RMSE']
        'RMSE'
        'MAE'
        'Quantile:alpha=value'
        'LogLinQuantile:alpha=value'
        'Poisson'
        'MAPE'
    """
    def __init__(
        self,
	detailed_profile=True,
        iterations=None,
`
where can I find  the logs? At console a have an ordinary messages, nothing special

**Comment by Noxoomo** [comment]
what logging_level did you use? should be logging_level: Info (or Debug) + detailed_profile:True

**Comment by SergeyBridge** [comment]
@Noxoomo Hello, Vasily! 
Here is attached logs with logging_level=Info + detailed_profile:True

[catboostLogsGPU.txt](https://github.com/catboost/catboost/files/1802907/catboostLogsGPU.txt)

Hope  it will help you

**Comment by Noxoomo** [comment]
I afraid it could be caused by small pool size: there just 46k samples and it's just not enough to fully utilize GPU.

You could check, if your task needs default options (which are slow), or quality is good for boosting_type: Plain, border_count: 32. With this ones learning should run 4-8 times faster on GPU, and ≈1.5-2 times faster on CPU

**Comment by SergeyBridge** [comment]
@Noxoomo I tried 
GPU mode boosting_type: Plain, border_count: 32, 
it doesn't bring me any profit in comparing with CPU mode, default boosting_type:Ordered, default border_count:128 and rsm:0.1 

The same time for training, similar( a little better for GPU) result.

I don't have more ideas about this topic, Geforce GTX480  GPU mode for catboost right now. Everyting is working, but there is no profit in training speed. But I can use my computer when model is training with GPU, this is profit )))

**Comment by Noxoomo** [comment]
rsm 0.1 gives CPU a big advantage in speed (it calculates only 10% of features on each iterations).

**Comment by SergeyBridge** [comment]
yes, sadly, rsm is a great parameter, because no such big difference in training accuracy, in my case RMSE 16.42 vs 15.94, 1.03..., 3% worse and 10 times faster.

**Comment by annaveronika** [comment]
If you use rsm, you need to run for more iterations, this should improve the problem with accuracy. For rsm 0.1 we recommend around +20% of iterations.

**Comment by SergeyBridge** [comment]
I tried to ask about another  parameter, learning_rate, which is a gradient step. I can set it now into a constant value only, but would be great to reduce it during iterations, step by step. Linear or exponentially, etc.  Sorry, if it is off topic.

**Comment by Noxoomo** [comment]
> yes, sadly, rsm is a great parameter, because no such big difference in training accuracy, in my case RMSE 16.42 vs 15.94, 1.03..., 3% worse and 10 times faster.

btw, we have plans for rsm on GPU, but i don't think it'll be ready before summer

**Comment by annaveronika** [comment]
@SergeyBridge We don't have a plan to support this. We did a lot of experiments here, and it's always worse than setting lower learning rate. There are better ways to improve speed.
But if you think this feature is important, we are welcoming people to contribute!

**Comment by SergeyBridge** [comment]
@annaveronika What does it mean +20% ? If I use use_best_model:True and od_type:Iter, ? I should wait for 20% of what staff?

**Comment by annaveronika** [comment]
You initially set iterations=2000. If optimal number of iterations was 2000, then optimal number of iterations with rsm 0.1 will be greater - I suggest it'll be around 2400.

**Comment by SergeyBridge** [comment]
@annaveronika Ok, thanks )))
